### PR TITLE
Front: fix typescript configuration and then compilation errors

### PR DESCRIPTION
**Pull request by sim51**
Typescript configuration was bad, the configuration `noImplicitUseStrict` cannot be used when `alwaysStrict` is set to `true`.
TS compilation was not working, with a silent error when doing `yarn start`...

By removing this configuration, the project was not able to compile anymore (due to bad/missing TS types).
I fixed a lot of errors (sometimes by adding the `any` type which is not good).

FYI, the file `ChartHelpers.ts` needs to be reviewed.

NB: This PR is essential for me to make the upgrade to map-libre.

### CODE REVIEW

**Comment by multun** [comment]
@sim51 new changes on `dev` generated conflicts, can you address these?

**Comment by sim51** [comment]
I will do it this morning

File: front/src/applications/carto/Map.tsx [hunk 1]
```diff
@@ -82,16 +87,13 @@ function Map() {
...
-        ...viewport,
-        latitude: parseFloat(urlLat),
-        longitude: parseFloat(urlLon),
-        zoom: parseFloat(urlZoom),
-        bearing: parseFloat(urlBearing),
-        pitch: parseFloat(urlPitch),
-      });
-    }
+    const newViewport: Partial<Viewport> = {};
+    if (urlLat) newViewport.latitude = parseFloat(urlLat);
```
Comment (anisometropie) [inline front/src/applications/carto/Map.tsx]: We need to keep the `...viewport` or we lose the original data.
I propose of a lighter syntax :

```
updateViewportChange({
        ...viewport,
        ...(urlLat && { latitude: parseFloat(urlLat) }),
        ...(urlLon && { longitude: parseFloat(urlLon) }),
        ...(urlZoom && { zoom: parseFloat(urlZoom) }),
        ...(urlBearing && { bearing: parseFloat(urlBearing) }),
        ...(urlPitch && { pitch: parseFloat(urlPitch) }),
});
```

**Comment by sim51** [comment]
@multun It's done

File: front/src/applications/editor/tools/selection/components.tsx [hunk 2]
```diff
@@ -97,6 +97,7 @@ export const SelectionLeftPanel: FC = () => {
     selection.forEach((item) => types.add(item.objType));
 
     if (selection.length === 1) {
+      console.log('Editform', selection[0]);
```
Comment (anisometropie) [inline front/src/applications/editor/tools/selection/components.tsx]: to remove

File: front/src/applications/osrd/components/Helpers/ChartHelpers.ts [hunk 3]
```diff
@@ -268,12 +305,12 @@ export const interpolateOnPosition = (dataSimulation, keyValues, positionLocal)
 
 // Interpolation of cursor based on time position
 export const interpolateOnTime = (
-  dataSimulation: Regime,
+  dataSimulation: { [key: string]: any } | undefined,
```
Comment (anisometropie) [inline front/src/applications/osrd/components/Helpers/ChartHelpers.ts:308]: Regime is correct here.
listValues is an `Array<keyof Regime>`

for example a call  in `src/applications/osrd/views/OSRDSimulation/Map.tsx`
Screenshot_20221108_130854

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/applications/carto/Map.tsx (hunk 1 above)
Comment (sim51) [reply front/src/applications/carto/Map.tsx]: The original value is already in the state, no need to keep it.
In the map reducer, we merge the new viewport with the previous one.

File: front/src/applications/editor/tools/selection/components.tsx (hunk 2 above)
Comment (sim51) [reply front/src/applications/editor/tools/selection/components.tsx]: already done

File: front/src/applications/osrd/components/Helpers/ChartHelpers.ts (hunk 3 above)
Comment (sim51) [reply front/src/applications/osrd/components/Helpers/ChartHelpers.ts:308]: `interpolateOnTime` is also used here : https://github.com/DGEXSolutions/osrd/blob/dev/front/src/reducers/osrdsimulation/index.ts#L323 which is a `SimulationTrain` not a Regime.

**Review by jacomyal — APPROVED** [review]

**Review by anisometropie — APPROVED** [review]

**Pull request merged**